contracts/Governance.sol

```diff
@@ -159,11 +163,24 @@ contract Governance is Context, Ownable, ERC165, EIP712 {
         _;
     }
 
-    constructor(
+    function initialize(
         ISnapshotVoting token_,
         TimelockController timelock_,
-        address votesOracle_
-    ) EIP712(name(), version()) {
+        address votesOracle_,
+        address treasury_
+    ) external initializer {
+        require(
+            _treasury == address(0) && treasury_ != address(0),
+            "Governance::setInitialTreasury: treasury was set or new address is zero"
+        );
+        __Ownable_init();
+        __ERC165_init_unchained();
```

Comment:
Agree. But here is the blocker, when I removed unchained for both, Governance.sol exceed EIP maximum size by ~100 bytes, seems like it adds quite a bit of code in bytes. It's true that Governance.sol got quite large and I might need to find the ways to reduce it or cut some stuff, or maybe split into contracts (though not sure how).

Reply:
Can be solved by removing trivial checks like votersOracle != address(0) and similar ones. It's possible as those are changed only when executed through governance proposals and 99% proposal will not pass if turns out addresses to update in those proposals are 0.

Reply:
The other option would be reducing the require message sizes. 

Solhint does a check (default of warning) when the require message size over 32 bytes. Only done a bit of read and can't find anything authoritative on whether that still applies to the latest Solidity, but if you're really need to trim the bytes 🤷

Reply:
Yeah I would keep those checks or even add more advanced checks.

I would re-structure the contract into 2 instead of walking on the brink of the size limit.

Reply:
Re-structuring is a large problem, not for this PR, potentially for several PRs actually. I will work on it a bit later.

Reply:
> Re-structuring is a large problem, not for this PR, potentially for several PRs actually. I will work on it a bit later.

question: @bitparadigm have you made a GitHub issues for that? 

Creating the issue and posting the link in this comment, helps with closing off the branch of discussion on restructuring the contract in this PR.

Reply:
https://github.com/windranger-io/windranger-governance/issues/85 good point.